types/src/execution_result.rs

```diff
@@ -287,39 +266,54 @@ impl FromBytes for ExecutionResult {
     }
 }
 
-/// The effect of executing a single deploy.
+/// The journal of execution transforms from a single deploy.
 #[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Default, Debug)]
 #[cfg_attr(feature = "std", derive(JsonSchema))]
 #[serde(deny_unknown_fields)]
-pub struct ExecutionEffect {
+pub struct JsonExecutionJournal {
```

Comment:
I realize that it seems to be the approach we have now where the `Json*` representations of the specific type/ADT are defined next to the original type but I think these presentation types should be living in another module. The reasoning being that they are from a different bounded context - namely API/event stream/etc. and can evolve independently from the types they map from.

/cc @Fraser999